From 4b281805a5d87efdd406e1d59bf3abacbfa0f49a Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Wed, 09 Sep 2020 07:25:20 +0300
Subject: [PATCH] [SW-19846] goya2: i2cslave: debug mode address

In case of goya2, debug mode address of i2c slave
is 0x6A in case of secure boot. It will remain 0x47
in case of goya2 legacy flow. This patch adds
changes so each platform can have its own i2c slave
debug mode address.

Change-Id: I1b460eb726aa599e39445ede15b53874e61a3cf9
---

diff --git a/common/module_i2c_slave.c b/common/module_i2c_slave.c
index 9ba3c9c..aed5075 100644
--- a/common/module_i2c_slave.c
+++ b/common/module_i2c_slave.c
@@ -21,8 +21,6 @@
 #include "common.h"
 #include "module_i2c_slave.h"
 
-#define I2C_SLAVE_ADDRESS_DBG		0x47
-
 static struct module_i2c_data data;
 
 static void hl_i2c_slave_init(bool debug_mode)
diff --git a/gaudi/common.h b/gaudi/common.h
index ffb0b91..3deb131 100644
--- a/gaudi/common.h
+++ b/gaudi/common.h
@@ -38,6 +38,7 @@
 
 #define FUSE_WORDS_PER_BANK		64
 #define I2C_SLAVE_ADDRESS		0x4f
+#define I2C_SLAVE_ADDRESS_DBG		0x47
 
 #ifndef __ASSEMBLY__
 #include <spi.h>
diff --git a/goya/common.h b/goya/common.h
index fa8481f..7bf6f9c 100644
--- a/goya/common.h
+++ b/goya/common.h
@@ -25,3 +25,4 @@
 
 #define FUSE_WORDS_PER_BANK	0x10
 #define I2C_SLAVE_ADDRESS	0x4f
+#define I2C_SLAVE_ADDRESS_DBG	0x47
diff --git a/goya2/common.h b/goya2/common.h
index 6cd52a8..8dd1ae8 100644
--- a/goya2/common.h
+++ b/goya2/common.h
@@ -16,3 +16,4 @@
 
 #define FUSE_WORDS_PER_BANK		0x40
 #define I2C_SLAVE_ADDRESS		0x6A
+#define I2C_SLAVE_ADDRESS_DBG		0x47
diff --git a/goya2_arc/common.h b/goya2_arc/common.h
index 1805014..8caef64 100644
--- a/goya2_arc/common.h
+++ b/goya2_arc/common.h
@@ -20,6 +20,7 @@
 
 #define FUSE_WORDS_PER_BANK	0x40
 #define I2C_SLAVE_ADDRESS	0x6A
+#define I2C_SLAVE_ADDRESS_DBG	I2C_SLAVE_ADDRESS
 
 #define EFUSE_SYSCFG_DISABLE_TWI_IF_MASK        BIT(19)
 #define EFUSE_SYSCFG_FW_VERSION_LOCATION_SPI	BIT(14)
diff --git a/goya2_arc_gnu/common.h b/goya2_arc_gnu/common.h
index 1a3eeff..a42e133 100644
--- a/goya2_arc_gnu/common.h
+++ b/goya2_arc_gnu/common.h
@@ -21,6 +21,7 @@
 
 #define FUSE_WORDS_PER_BANK	0x40
 #define I2C_SLAVE_ADDRESS	0x6A
+#define I2C_SLAVE_ADDRESS_DBG	I2C_SLAVE_ADDRESS
 
 #define RECOVERY_TEXT_BASE	(0x0 + 0x4)
 #define RECOVERY_SIZE		0x4000
